src/client/containers/ProductPage/ProductPage.Container.js

```diff
@@ -1,12 +1,87 @@
+/* eslint-disable no-console */
+/* eslint-disable radix */
 import React from 'react';
+import { useParams } from 'react-router-dom';
+import { ProductDetails } from '../../components/ProductDetails/ProductDetails.component';
+import Carousel from '../../components/Carousel/Carousel.component';
+import ButtonComponent from '../../components/Button/Button.component';
+import Loader from '../../components/Loader/index';
 import './ProductPage.Style.css';
+import { useFetchApi } from '../../hooks/UseFetchApi';
 
 const ProductPageContainer = () => {
+  const { id } = useParams();
+  const [product, setProduct] = React.useState({});
+  const [category, setCategory] = React.useState('');
+  const [similarProduct, setSimilarProduct] = React.useState([]);
+
+  const productData = useFetchApi(`products/${id}`);
+
+  React.useEffect(() => {
```

Comment:
If you want to go thurther in clearing your code you can delete this.
You don`t really need this useEffect you are duplicating logic inside useFetchHook - productData already saved in useFetch state, and you can rename it to whatever using object destructuring So All you page can look like this

```
const ProductPageContainer = () => {
  const { id } = useParams();

  const {  data: [product],   productLoading } = useFetchApi(`products/${id}`); //this will take first element from data array and rename it to product
  const {  data: [category],  categoryLoading,  } = useFetchApi(`categories/${product?.category_id}`); //this will take first element from data array and rename it to category
  const { data: similarProduct, similarProductsLoading } = useFetchApi(`products?category=${category?.name}`);

  const addToCartHandler = () => {
   .....
  };
  const exploreCategoryHandler = () => {
  ....
  };

  return (
    <div>
      <div>
        <div className="p-detail">
          {productLoading ? (
            <Loader />
          ) : (
            <ProductDetails
              imgSource={product?.picture}
              ProductName={product?.name}
              RemainingUnit={parseInt(product?.stock_amount)}
              Price={parseInt(product?.price)}
              productColor={product?.color}
              productSize={product?.size}
              onClick={addToCartHandler}
            />
          )}
        </div>
        <div className="similar-product">
          <h1>SIMILAR PRODUCT</h1>
          {categoryLoading || similarProductsLoading ? (
            <Loader />
          ) : (
            <Carousel imageArray={similarProduct.map((item) => item.picture)} />
          )}
        </div>
        <div className="explore-btn">
          <ButtonComponent
            title="EXPLORE THIS CATEGORY"
            onClick={exploreCategoryHandler}
            backgroundColor="gray"
          />
        </div>
      </div>
    </div>
  );
};
export default ProductPageContainer;
```

Reply:
Hi julia,
have you run your code?? is it working.. 
i tried yesterday same like that .. its not working ...Thats why i did like this ...

Reply:
Yes i tried it and it is working

Reply:
Hi Guys, I got approval from Daniel so I merged this PR.